tasks/ui_framework.js

```diff
@@ -77,27 +82,46 @@ module.exports = function (grunt) {
   }
 
   function uiFrameworkCompile() {
-    sass.render({
-      file: 'ui_framework/components/index.scss'
-    }, function (error, result) {
-      if (error) {
-        grunt.log.error(error);
-      }
-
-      postcss([postcssConfig])
-        .process(result.css, { from: 'ui_framework/components/index.scss', to: 'ui_framework/dist/ui_framework.css' })
-        .then(result => {
-          grunt.file.write('ui_framework/dist/ui_framework.css', result.css);
-
-          if (result.map) {
-            grunt.file.write('ui_framework/dist/ui_framework.css.map', result.map);
-          }
-        });
+    return new Promise(resolve => {
+      sass.render({
+        file: 'ui_framework/components/index.scss'
+      }, function (error, result) {
+        if (error) {
+          grunt.log.error(error);
+        }
+
+        postcss([postcssConfig])
+          .process(result.css, { from: 'ui_framework/components/index.scss', to: 'ui_framework/dist/ui_framework.css' })
+          .then(result => {
+            grunt.file.write('ui_framework/dist/ui_framework.css', result.css);
+
+            if (result.map) {
+              grunt.file.write('ui_framework/dist/ui_framework.css.map', result.map);
+            }
+
+            resolve();
+          });
+      });
     });
   }
 
   function uiFrameworkWatch() {
-    const debouncedCompile = debounce(uiFrameworkCompile, 400, { leading: true });
+    const debouncedCompile = debounce(() => {
+      // Compile the SCSS in a separate process because node-sass throws a fatal error if it fails
+      // to compile.
+      grunt.util.spawn({
```

Comment:
If you wanted to use a promise here you could as well, but I feel the callback is fine if you want to leave it.

```
const spawn = promisify(grunt.util.spawn);

...

spawn({
  cmd: isPlatformWindows ? '.\\node_modules\\.bin\\grunt.cmd' : './node_modules/.bin/grunt',
  args: [ 'uiFramework:compileCss' ]
}).then(grunt.log.writeln).catch(e => grunt.log.error(e.stdout));
```